src/topology/bounded_continuous_function.lean

```diff
@@ -429,5 +429,90 @@ def of_normed_group_discrete {α : Type u} {β : Type v}
   (f : α  → β) (C : ℝ) (H : ∀x, norm (f x) ≤ C) : α →ᵇ β :=
 of_normed_group f C H continuous_of_discrete_topology
 
+/- Formula for norm. -/
+lemma norm_eq (f : α →ᵇ β) :
+  ∥f∥ = Inf {C : ℝ | 0 ≤ C ∧ ∀ (x : α), ∥f x∥ ≤ C} :=
+begin
+  rw [← sub_zero f, ← dist_eq_norm, bounded_continuous_function.dist_eq],
+  simp,
+end
+
 end normed_group
+
+section normed_space
+/-! In this section, if `β` is a normed space, then we show that the space of bounded
+continuous functions from `α` to `β` inherits a normed space structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables {f g : α →ᵇ β} {x : α} {C : ℝ}
+
+instance : has_scalar 𝕜 (α →ᵇ β) :=
+⟨λc, λf, ⟨λx, c • f.1 x,
+  continuous.smul continuous_const f.2.left,
+  begin
+    cases f.2.right with C hbound,
+    use ∥c∥ * C,
+    intros,
+    have hnneg : 0 ≤ ∥c∥ := norm_nonneg c,
+    specialize hbound x y,
+    rw dist_eq_norm at hbound ⊢,
+    calc ∥c • f x - c • f y∥ = ∥c • (f x - f y)∥ : by rw smul_sub c (f x) (f y)
+    ... = ∥c∥ * ∥f x - f y∥ : norm_smul c (f x - f y)
+    ... ≤ ∥c∥ * C : mul_le_mul_of_nonneg_left hbound hnneg
+  end⟩⟩
+
+instance : module 𝕜 (α →ᵇ β) :=
+  module.of_core $
+  { smul     := (•),
+    smul_add := λ c f g, ext $ λ x, smul_add c (f x) (g x),
+    add_smul := λ c₁ c₂ f, ext $ λ x, add_smul c₁ c₂ (f x),
+    mul_smul := λ c₁ c₂ f, ext $ λ x, mul_smul c₁ c₂ (f x),
+    one_smul := λ f, ext $ λ x, one_smul 𝕜 (f x) }
+
+instance : vector_space 𝕜 (α →ᵇ β) :=
+{ .. bounded_continuous_function.module }
+
+lemma bounded_continuous_sub_smul (c : 𝕜) (f : α →ᵇ β) : ∥c • f∥ ≤ ∥c∥ * ∥f∥ :=
+begin
+  have hnneg : 0 ≤ ∥ c ∥ := norm_nonneg c,
+  rw norm_eq (c • f),
+  apply real.Inf_le,
+  { use 0, intros y hy,
+    rw set.mem_set_of_eq at hy,
+    exact hy.left },
+  { rw set.mem_set_of_eq,
+    split,
+    { exact mul_nonneg' hnneg (norm_nonneg f) },
+    { intros,
+      calc ∥(c • f) x∥ = ∥c∥ * ∥f x∥ : norm_smul c (f x)
+      ... ≤ ∥c∥ * ∥f∥ : mul_le_mul_of_nonneg_left (norm_coe_le_norm x) hnneg } }
+end
+
+lemma bounded_continuous_smul (c : 𝕜) (f : α →ᵇ β) : ∥c • f∥ = ∥c∥ * ∥f∥ :=
+begin
+  by_cases h : c = 0,
+  { rw [h, zero_smul, norm_zero, norm_zero, zero_mul] },
+  { have hnneg : 0 ≤ ∥c∥ := norm_nonneg c,
+    apply le_antisymm,
+    { exact bounded_continuous_sub_smul c f },
+    { have hinv : ∥f∥ ≤ ∥1 / c∥ * ∥c • f∥,
+      { calc ∥f∥ = ∥(1 : 𝕜) • f∥ : by rw one_smul
+        ... = ∥(1 / c * c ) • f∥ : by rw (div_mul_cancel 1 h)
+        ... = ∥(1 / c) • ( c • f)∥ : by rw ← (mul_smul _ _ _)
+        ... ≤ ∥1 / c∥ * ∥c • f∥ : bounded_continuous_sub_smul (1 / c) (c • f) },
```

Comment:
```suggestion
    { exact f.smul_le_mul c },
    { have hinv : ∥f∥ ≤ ∥1 / c∥ * ∥c • f∥,
      { calc ∥f∥ = ∥(1 : 𝕜) • f∥ : by rw one_smul
        ... = ∥(1 / c * c ) • f∥ : by rw (div_mul_cancel 1 h)
        ... = ∥(1 / c) • ( c • f)∥ : by rw ← (mul_smul _ _ _)
        ... ≤ ∥1 / c∥ * ∥c • f∥ : (c • f).smul_le_mul (1 / c) },
```
Just using the new name for the lemma above this one and dot notation.

Reply:
This is nice!